### PR TITLE
Wait for relocations and disk threshold monitor in DiskThresholdDeciderIT

### LINKED ISSUE #62326: DiskThresholdDeciderIT.testHighWatermarkNotExceeded failure
**Build scan**:
https://gradle-enterprise.elastic.co/s/okz2lziucuxkq/tests/:server:internalClusterTest/org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT/testHighWatermarkNotExceeded

**Repro line**:
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded" -Dtests.seed=60A6AF6A936EF834 -Dtests.security.manager=true -Dtests.locale=ro-RO -Dtests.timezone=America/Pangnirtung -Druntime.java=11
```

**Reproduces locally?**: no

**Applicable branches**: master

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.buildToolTypes=gradle&search.buildToolTypes=maven&search.relativeStartTime=P7D&search.timeZoneId=Europe/London&tests.container=org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT&tests.sortField=FAILED&tests.test=testHighWatermarkNotExceeded&tests.unstableOnly=true

**Failure excerpt**:
```
java.lang.AssertionError: |  
  | Expected: a collection with size <1> |  
  | but: collection size was <0>

at __randomizedtesting.SeedInfo.seed([60A6AF6A936EF834:89874ED813A831DA]:0) |  
  |   | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  |   | at org.junit.Assert.assertThat(Assert.java:956) |  
  |   | at org.junit.Assert.assertThat(Assert.java:923) |  
  |   | at org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded(DiskThresholdDeciderIT.java:160) |  
  |  

```

### PR DESCRIPTION
**Pull request by tlrx**
Sadly I did not manage to reproduce #62326  on any of the computers I have, but the following log lines of the last CI failure makes me think that it is possible that the DiskThresholdMonitor is being processing a `reroute` (`checkInProgress` is `true`) after the last cluster update that detected that the disk has enough free space again to hold shard:

```
1> [2020-09-14T10:52:38,949][INFO ][o.e.c.r.a.d.DiskThresholdDeciderIT] [testHighWatermarkNotExceeded] Index [5067] docs async: [false] bulk: [true] partitions [6]
  1> [2020-09-14T10:52:42,460][INFO ][o.e.c.r.a.DiskThresholdMonitor] [node_t0] skipping monitor as a check is already in progress
  1> [2020-09-14T10:52:42,462][INFO ][o.e.c.r.a.DiskThresholdMonitor] [node_t0] high disk watermark [10kb] exceeded on [zlTDZ1KWRIOAD_Xs6J1-gA][node_t2][/dev/shm/elastic+elasticsearch+master+multijob+fast+part1/server/build/testrun/internalClusterTest/temp/org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT_60A6AF6A936EF834-001/tempDir-003/node-2] free: 0b[0%], shards will be relocated away from this node; currently relocating away shards totalling [75577] bytes; the node is expected to be below the high disk watermark when these relocations are complete
  1> [2020-09-14T10:52:42,710][INFO ][o.e.c.r.a.DiskThresholdMonitor] [node_t0] skipping monitor as a check is already in progress
  1> [2020-09-14T10:52:42,906][INFO ][o.e.c.r.a.DiskThresholdMonitor] [node_t0] skipping monitor as a check is already in progress
  1> [2020-09-14T10:52:43,071][WARN ][o.e.c.r.a.DiskThresholdMonitor] [node_t0] high disk watermark [10kb] exceeded on [zlTDZ1KWRIOAD_Xs6J1-gA][node_t2][/dev/shm/elastic+elasticsearch+master+multijob+fast+part1/server/build/testrun/internalClusterTest/temp/org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT_60A6AF6A936EF834-001/tempDir-003/node-2] free: 7.1kb[8.7%], shards will be relocated away from this node; currently relocating away shards totalling [0] bytes; the node is expected to continue to exceed the high disk watermark when these relocations are complete
  1> [2020-09-14T10:52:43,096][WARN ][o.e.c.r.a.DiskThresholdMonitor] [node_t0] high disk watermark [10kb] exceeded on [zlTDZ1KWRIOAD_Xs6J1-gA][node_t2][/dev/shm/elastic+elasticsearch+master+multijob+fast+part1/server/build/testrun/internalClusterTest/temp/org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT_60A6AF6A936EF834-001/tempDir-003/node-2] free: 7.1kb[8.7%], shards will be relocated away from this node; currently relocating away shards totalling [0] bytes; the node is expected to continue to exceed the high disk watermark when these relocations are complete
  1> [2020-09-14T10:52:43,098][INFO ][o.e.c.r.a.d.DiskThresholdDeciderIT] [testHighWatermarkNotExceeded] [DiskThresholdDeciderIT#testHighWatermarkNotExceeded]: cleaning up after test
```

This pull request introduces some assertBusy() so that it gives a bit more time for the shards and the DiskThreshMonitor's reroute to be processed.

Closes #62326

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java [hunk 1]
```diff
@@ -151,13 +150,17 @@ public void testHighWatermarkNotExceeded() throws InterruptedException {
         // reduce disk size of node 0 so that no shards fit below the high watermark, forcing all shards onto the other data node
         // (subtract the translog size since the disk threshold decider ignores this and may therefore move the shard back again)
         fileSystemProvider.getTestFileStore(dataNode0Path).setTotalSpace(minShardSize + WATERMARK_BYTES - 1L);
-        refreshDiskUsage();
-        assertThat(getShardRoutings(dataNode0Id), empty());
+        assertBusy(() -> {
+            refreshDiskUsage();
```
Comment (original-brownbear) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java:154]: `refreshDiskUsage` triggers a reroute, shouldn't we have that outside the busy assert and just make sure the the CS application is fully processed by busy asserting on the line below only? That should be all that's needed?

**Review by original-brownbear — COMMENTED** [review]
One question, but I think this is what's breaking things indeed

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java (hunk 1 above)
Comment (tlrx) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java]: I agree, I think you're right.

**Review by original-brownbear — APPROVED** [review]
LGTM thanks Tanguy!

**Comment by tlrx** [comment]
Thanks Armin

**Pull request merged**